daemon/list.go

```diff
@@ -499,14 +499,14 @@ func includeContainerInList(container *container.Container, ctx *listContext) it
 	networkExist := fmt.Errorf("container part of network")
 	if ctx.filters.Include("network") {
 		err := ctx.filters.WalkValues("network", func(value string) error {
-			if _, ok := container.NetworkSettings.Networks[value]; ok {
-				return networkExist
-			}
```

Comment:
I think we can keep this shortcut, so that we don't have to loop the list of networks if there's a direct match